src/Components/NoteInput/NoteInput.jsx

```diff
@@ -21,34 +29,48 @@ export function NoteInput() {
         e.stopPropagation();
         setInputData((prev) => ({ ...prev, bgColor: color }));
     }
+    function changeSelectedPriorityHandler(value) {
+        setInputData((prev) => ({ ...prev, selectedPriority: value }));
+    }
+    function falseStatesSetter() {
+        setToggleClrPallette(false);
+        setToggleTagModal(false);
+        setTogglePriority(false);
+    }
+    function resetFunction() {
+        setInputData(initialData);
+        falseStatesSetter();
+        tagsDispatchFunction({ type: "RESET_GLOBAL_TAG_STATES" });
+    }
+    function setCurrentFieldhandler(e, setterFunction) {
+        e.stopPropagation();
+        falseStatesSetter();
+        setterFunction(true);
+    }
```

Comment:
Not mandatory but extracting these functions in the hook will increase the overall readability of the code.